app/controllers/api/v1/sources_controller.rb

```diff
@@ -15,6 +18,49 @@ def create
 
         render :json => source, :status => :created, :location => instance_link(source)
       end
+
+      def check_availability
+        source = Source.find(params[:source_id])
+
+        Sources::Api::Messaging.client.publish_topic(
+          :service => "platform.topological-inventory.operations-#{source.source_type.name}",
+          :event   => "Source.availability_check",
+          :payload => {
+            :params => {
+              :source_id       => source.id.to_s,
+              :external_tenant => source.tenant.external_tenant
+            }
+          }
+        )
+
+        check_application_availability(source)
+
+        render :json => source, :status => :accepted
+      end
+
+      private
+
+      def check_application_availability(source)
+        source.application_types.each do |app_type|
+          url = app_type.availability_status_url
+          next if url.blank?
+
+          headers = {
+            "Content-Type"  => "application/json",
+            "x-rh-identity" => Base64.strict_encode64({'identity' => { 'account_number' => source.tenant.external_tenant }}.to_json)
+          }
+
+          uri = URI.parse(url)
+          net_http = Net::HTTP.new(uri.host, uri.port)
+          request  = Net::HTTP::Post.new(uri.request_uri, headers)
+          request.body = { "source_id" => source.id.to_s }.to_json
```

Comment:
/cc @agrare just initial prototype here, we need to design/agree on signature/contract with the apps.

Reply:
:+1: this is what we discussed with @dccurtis